source/allocator.c

```diff
@@ -25,30 +25,65 @@
 #    pragma warning(disable : 4100)
 #endif
 
+#ifndef PAGE_SIZE
+#    define PAGE_SIZE (4 * 1024)
+#endif
+
 bool aws_allocator_is_valid(const struct aws_allocator *alloc) {
     /* An allocator must define mem_acquire and mem_release.  All other fields are optional */
     return alloc && AWS_OBJECT_PTR_IS_READABLE(alloc) && alloc->mem_acquire && alloc->mem_release;
 }
 
 static void *s_default_malloc(struct aws_allocator *allocator, size_t size) {
     (void)allocator;
-    return malloc(size);
+    /* larger allocations should be aligned so that AVX and friends can run
+     * the aligned versions of memcpy and memset and friends on big buffers */
+    const size_t alignment = sizeof(void *) * (size > PAGE_SIZE ? 8 : 1);
```

Comment:
Can we add a comment for us mere mortals explicitly giving out the contract:

64 byte alignment for > page allocations on 64 bit systems
32 byte alignment for > page allocations on 32 bit systems
8 byte alignment for <= page allocations on 64 bit systems
4 byte alignment for <= page allocations on 32 bit systems

At least that's that the code does.  I'd feel more confident in the correctness if the comment echoed that contract.

Reply:
Good call.